### PR TITLE
Add transfer metadata form

### PR DESCRIPTION
**Pull request by rhernand3z**
Adds the transfer form react component with validation
Adds the node/express middleware to handle SQS data
Adds documentation for AWS SQS

Fixes: #3

### CODE REVIEW

File: src/server/routes/api/index.js
Comment (ktp242) [inline src/server/routes/api/index.js]: I feel we can test `isEmpty(barcodes) || typeof protectCGD !== 'boolean' || isEmpty(action) || isEmpty(email)` together before we get into the condition check for `type` since they are necessary in both `update` and `transfer`

Comment (rhernand3z) [reply src/server/routes/api/index.js]: that is very true, good catch -- I will simplify the logic. Thanks kTp~

File: src/server/routes/api/index.js
Comment (ktp242) [inline src/server/routes/api/index.js]: If the 'type' is invalid, it suggests our server is unable to to assign a proper `action` to it (since neither the user or AWS SQS is involved to put in the `action`), so I think the error code should be `500`

**Comment by ktp242** [comment]
Are the functions to display form errors working on this PR or still being built? I am not able to get errors to display on my UI locally.

**Comment by rhernand3z** [comment]
@ktp242 those are in the PR under renderFormSubmissionResults()

File: src/server/routes/api/index.js
Comment (rhernand3z) [reply src/server/routes/api/index.js]: I spoke about it with @thisisstephenbetts -- any 5xx range error is considered a server error. An internal server error makes sense as well, I don't mind changing it to that.

**Comment by ktp242** [comment]
Does it mean I won't get error block on this branch if I submit an empty form? And the submitting is actually not work here either?

File: src/server/routes/api/index.js
Comment (ktp242) [reply src/server/routes/api/index.js]: I see. That's cool! It is just a suggestion. Final decision is of course up to you and @thisisstephenbetts!

**Comment by rhernand3z** [comment]
You can't submit the form, there is JS validation that prohibits an empty form submit. Right now toolkit was just updated with textarea required fields for error and success, that has not been integrated and is not in the scope of this ticket. 

You can submit a valid form, if there is an API error... you will get an error message in the page - that is where renderFormSubmissionResults() comes in. 

Individual form field ui representation will be part of next sprint with the toolkit. Hope that clears things up, I did not want to wait on toolkit to add this as a huge pr.

**Comment by ktp242** [comment]
Got you! Thanks for the response. Just want to make sure which functions I need to review on the PR.

File: src/server/routes/api/index.js
Comment (rhernand3z) [inline src/server/routes/api/index.js:20]: @ktp242 -- updated, simplified the logic checks and returned a better response object to easily debug

**Review by ktp242 — APPROVED** [review]

File: src/server/routes/api/index.js
Comment (ktp242) [reply src/server/routes/api/index.js:20]: 👍

**Pull request merged**
